### PR TITLE
feat: Migrate from Groovy to Kotlin Build configuration 

### PR DESCRIPTION
**Pull request by angrezichatterbox**
### Contributor checklist

-   [x] This pull request is on a [separate branch](https://docs.github.com/en/get-started/quickstart/github-flow) and not the main branch

---

### Description

- The pull request migrates the Build configuration files from Groovy to Kotlin. 
- Kapt has been changed to ksp as it was showing up as warnings to Migrate from kapt to ksp. 
### Related issue

-   #80

### CODE REVIEW

**Comment by angrezichatterbox** [comment]
It seems detekt did not have a source earlier. Now detekt seems to work properly.  Should I disable those for now?

**Comment by andrewtavis** [comment]
Yes, @angrezichatterbox, sounds good. Would be great if you disable the rules and make good first issues for enabling them and fixing all of their errors 😊

**Comment by angrezichatterbox** [comment]
> Yes, @angrezichatterbox, sounds good. Would be great if you disable the rules and make good first issues for enabling them and fixing all of their errors 😊

Yes I will disable the rules and make all the issues together :).

**Comment by andrewtavis** [comment]
Let me know when this is ready for a final review, @angrezichatterbox :)

**Comment by angrezichatterbox** [comment]
> Let me know when this is ready for a final review, @angrezichatterbox :)

There seem to be a lot that detekt seems to find. I will add them all and then comment in here.

**Comment by angrezichatterbox** [comment]
It is now ready for review. There are around 26 that I have disabled.

File: detekt.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
+style:
```
Comment (andrewtavis) [inline detekt.yml:1]: Nice, @angrezichatterbox :) Feel free to make good first issues for each of these that we want to be enabled!

Comment (angrezichatterbox) [reply detekt.yml:1]: I will make issues for those :).

**Comment by andrewtavis** [comment]
@angrezichatterbox, I'm getting the following error when trying to build:

```
Execution failed for task ':app:kspCoreDebugKotlin'.
> Inconsistent JVM-target compatibility detected for tasks 'compileCoreDebugJavaWithJavac' (1.8) and 'kspCoreDebugKotlin' (17).
```

Are you able to build the project on this branch?

**Comment by andrewtavis** [comment]
Note that you'll need to bring my changes down 😊

**Comment by angrezichatterbox** [comment]
> @angrezichatterbox, I'm getting the following error when trying to build:
> 
> ```
> Execution failed for task ':app:kspCoreDebugKotlin'.
> > Inconsistent JVM-target compatibility detected for tasks 'compileCoreDebugJavaWithJavac' (1.8) and 'kspCoreDebugKotlin' (17).
> ```
> 
> Are you able to build the project on this branch?

I have fixed this error. I had made some changes to try to get detekt running on my system. Sorry.

**Comment by andrewtavis** [comment]
No stress whatsoever, @angrezichatterbox! Will review it now :)

**Review by andrewtavis — APPROVED** [review]
App's running great and as noted before the build files are now much more straightforward :) Thanks for this, @angrezichatterbox! Really great that this is being handled pre Hacktoberfest 🎃

**Pull request merged**